src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1532,26 +1542,57 @@ private static PullRequestSCMRevision createPullRequestSCMRevision(GHPullRequest
         String baseHash = pr.getBase().getSha();
         String prHeadHash = pr.getHead().getSha();
         String mergeHash = null;
-        switch (prhead.getCheckoutStrategy()) {
-            case MERGE:
-                Connector.checkApiRateLimit(listener, github);
-                baseHash = ghRepository.getRef("heads/" +  pr.getBase().getRef()).getObject().getSha();
-                if (pr.getMergeable() != null) {
-                    mergeHash = Boolean.TRUE.equals(pr.getMergeable()) ? pr.getMergeCommitSha() : PullRequestSCMRevision.NOT_MERGEABLE_HASH;
+
+        if (prhead.isMerge()) {
```

Comment:
Before we would validate the merge later.  Now we do a bunch of work to _make_ the merge valid or to ignore it.